docs/simon.rst

```diff
@@ -0,0 +1,72 @@
+Simon's Algorithm
+=================
+
+Overview
+--------
+
+This module emulates Simon's Algorithm.
+
+Simon's problem is summarized as follows.
+A function \\(f :\\{0,1\\}^n\\rightarrow\\{0,1\\}^n\\) is promised to be either
+one-to-one, or two-to-one with some nonzero \\(n\\)-bit mask \\(s\\).
+The latter condition means that for any two different \\(n\\)-bit numbers
+\\(x\\) and \\(y\\), \\(f(x)=f(y)\\) if and only if \\(x\\oplus y = s\\). The problem
+then is to determine whether \\(f\\) is one-to-one or two-to-one, and, if
+the latter, what the mask \\(s\\) is, in as few queries to \\(f\\) as possible.
+
+The problem statement and algorithm can be explored further, at a high level,
+in reference [1]_. The implementation of the algorithm in this module,
+however, follows [2]_.
+
+
+Algorithm and Details
+---------------------
+This algorithm involves a quantum component and a classical component.
+The quantum part follows similarly to other blackbox oracle algorithms.
+First, assume a blackbox oracle \\(U_f\\) is available with the property
+$$U_f\\vert x\\rangle\\vert y\\rangle = \\vert x\\rangle\\vert y\\oplus f(x)\\rangle$$
+
+where the top \\(n\\) qubits are the input, and the bottom \\(n\\) qubits are called ancilla qubits.
+
+Prepare the input into the state
+
+$$\\vert +\\rangle^{\\otimes n}\\vert 0\\rangle^{\\otimes n}$$
```

Comment:
Is the notation |+> standard? It's used in the linked lecture notes, but it's not defined here. (It's the uniformly superimposed state vector.)